src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,278 @@
+  private static Map<String, String> createPropertyMap()
+  {
+    Map<String,String> propertyMapper = new HashMap<>();
+    propertyMapper.put("category", Param.ITEM_CATEGORY);
+    propertyMapper.put("product_id", Param.ITEM_ID);
+    propertyMapper.put("name", Param.ITEM_NAME);
+    propertyMapper.put("price", Param.PRICE);
+    propertyMapper.put("quantity", Param.QUANTITY);
+    propertyMapper.put("query", Param.SEARCH_TERM);
+    propertyMapper.put("shipping", Param.SHIPPING);
+    propertyMapper.put("tax", Param.TAX);
+    propertyMapper.put("total", Param.VALUE);
+    propertyMapper.put("revenue", Param.VALUE);
+    propertyMapper.put("order_id", Param.TRANSACTION_ID);
+    propertyMapper.put("currency", Param.CURRENCY);
+    return propertyMapper;
+  }
+
+  public FirebaseIntegration(Context context, Logger logger) {
+    this.logger = logger;
+    firebaseAnalytics = FirebaseAnalytics.getInstance(context);
+  }
+
+  @Override
+  public void onActivityResumed(Activity activity) {
+    super.onActivityResumed(activity);
 
-  private static final String FIREBASE_ANALYTICS_KEY = "Firebase Analytics";
+    PackageManager packageManager = activity.getPackageManager();
+    try {
+      ActivityInfo info =
+              packageManager.getActivityInfo(activity.getComponentName(), PackageManager.GET_META_DATA);
+      CharSequence activityLabel = info.loadLabel(packageManager);
+      /** setCurrentScreen should only be called in the onResume() activity */
+      firebaseAnalytics.setCurrentScreen(
+              activity, activityLabel.toString(), null /* class override */);
+      logger.verbose(
+              "firebaseAnalytics.setCurrentScreen(%s, %s, null /* class override */);",
+              activity, activityLabel.toString());
 
-  public FirebaseIntegration(Context context) {
-    mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
+    } catch (PackageManager.NameNotFoundException e) {
+      throw new AssertionError("Activity Not Found: " + e.toString());
+    }
   }
 
-  @Override public void onActivityStarted(Activity activity) {
-    super.onActivityStarted(activity);
+  @Override
+  public void identify(IdentifyPayload identify) {
+    super.identify(identify);
 
+    if (!isNullOrEmpty(identify.userId())) {
+      firebaseAnalytics.setUserId(identify.userId());
+      Map<String, Object> traits = identify.traits();
+      for (Map.Entry<String, Object> entry : traits.entrySet()) {
+        String trait = entry.getKey();
+        trait = trait.trim().replaceAll(" ", "_");
+        if (trait.length() > 40) {
+          trait = trimKey(trait);
+        }
+        String formattedValue;
+        if (entry.getValue() instanceof Date) {
+          Date value = (Date) entry.getValue();
+          formattedValue = formatDate(value);
+        } else if (entry.getValue() instanceof String && ISO_DATE.matcher(String.valueOf(entry.getValue())).matches()) {
```

Comment:
Not 100% sure if we should have this check here (we don't do this in other integrations). If a user sends a date, it should be a `Date` object and not a string.

Reply:
Good question. I put this check there because Firebase requires dates to be formatted as "YYYY-MM-DD" strings. In case a user decides to pass a date string rather than a date object, this check will ensure that the string is formatted correctly. Do you think this is appropriate, or doing too much validation?

Reply:
I would check a few other integrations to see how we handle it. I think it's better to restrict us to only using `Date` objects. Doing a regex check is much more expensive and likely to be less consistent between integrations (e.g. copy paste bugs).

Reply:
Looks like we don't do any special handling for dates in other Android SDKs at all, so I'll only reformat date objects to "YYYY-MM-DD" here and document that if a customer wants to send a date string for some reason, it must be formatted as "YYYY-MM-DD".

Reply:
> document that if a customer wants to send a date string for some reason, it must be formatted as "YYYY-MM-DD".

I think it's better for now then to make sure only `Date` objects are treated as dates.